src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/arrayBuilder.jsx

```diff
@@ -0,0 +1,443 @@
+  const navForwardFinishedItem = ({
+    goPath,
+    urlParams,
+    pathname,
+    pageList,
+  }) => {
+    let path = summaryPath;
+    if (urlParams?.edit) {
+      const index = getUrlPathIndex(pathname);
+      const review = reviewPath || pageList[pageList.length - 1]?.path;
+      const basePath = urlParams?.review ? review : summaryPath;
+      path = createArrayBuilderUpdatedPath({
+        basePath,
+        index,
+        nounSingular,
+      });
+    }
+    goPath(path);
+  };
+
+  /** @type {FormConfigPage['onNavBack']} */
+  const navBackFirstItem = onNavBackRemoveAddingItem({
+    arrayPath,
+    summaryPathUrl: summaryPath,
+  });
+
+  /** @type {FormConfigPage['onNavForward']} */
+  const navForwardSummary = ({ formData, goPath, pageList }) => {
+    if (formData[hasItemsKey]) {
+      const index = formData[arrayPath] ? formData[arrayPath].length : 0;
+      const path = createArrayBuilderItemAddPath({
+        path: itemFirstPagePath,
+        index,
+      });
+      goPath(path);
+    } else {
+      const nextPage = getPageAfterPageKey(pageList, itemLastPageKey);
+      goPath(nextPage?.path);
+    }
+  };
+
+  pageBuilder.summaryPage = pageConfig => {
+    const requiredOpts = ['title', 'path', 'uiSchema', 'schema'];
+    verifyRequiredPropsPageConfig('summaryPage', requiredOpts, pageConfig);
+
+    return {
+      CustomPageReview: () => SummaryCards,
+      onNavForward: navForwardSummary,
+      ...pageConfig,
+      uiSchema: {
+        ...pageConfig.uiSchema,
+        'ui:description': SummaryCards,
+      },
+    };
+  };
+
+  pageBuilder.itemFirstPage = pageConfig => {
+    const { depends, ...restPageConfig } = pageConfig;
+    const requiredOpts = ['title', 'path', 'uiSchema', 'schema'];
+    verifyRequiredPropsPageConfig('itemFirstPage', requiredOpts, pageConfig);
+
+    return {
+      ...commonItemPageConfig(depends),
+      onNavBack: navBackFirstItem,
+      onNavForward: onNavForwardKeepUrlParams,
+      ...restPageConfig,
+      ...commonItemSchemas(restPageConfig.uiSchema, restPageConfig.schema),
+    };
+  };
+
+  pageBuilder.itemMiddlePage = pageConfig => {
+    const { depends, ...restPageConfig } = pageConfig;
+    const requiredOpts = ['title', 'path', 'uiSchema', 'schema'];
+    verifyRequiredPropsPageConfig('itemMiddlePage', requiredOpts, pageConfig);
+
+    return {
+      ...commonItemPageConfig(depends),
+      onNavBack: onNavBackKeepUrlParams,
+      onNavForward: onNavForwardKeepUrlParams,
+      ...restPageConfig,
+      ...commonItemSchemas(restPageConfig.uiSchema, restPageConfig.schema),
+    };
+  };
+
+  pageBuilder.itemSinglePage = pageConfig => {
+    const { depends, ...restPageConfig } = pageConfig;
+    const requiredOpts = ['title', 'path', 'uiSchema', 'schema'];
+    verifyRequiredPropsPageConfig('itemSinglePage', requiredOpts, pageConfig);
+
+    return {
+      ...commonItemPageConfig(depends),
+      onNavBack: navBackFirstItem,
+      onNavForward: navForwardFinishedItem,
```

Comment:
Does a single page include a summary page? I guess it doesn't? So would navigating forward go to the next page in the form - outside of the array pages? I'm confused because the `itemLastPage` also uses this `navForwardFinishedItem` and I would expect that to go back to the summary page? In my head, the summary page navigates to the next page outside of the array page if you choose "No".

Reply:
in this pattern, there is always a summary page which you will return to after every iteration. so even a single item page, is still going summary -> item -> summary -> item if you want to keep going.